File: v2.x/Examples/SciChartDemo/SciChartDemo/DoubleSeries.m
//
//  DoubleSeries.m
//  SciChartDemo
//
//  Created by Admin on 19/04/2017.
//  Copyright © 2017 ABT. All rights reserved.
//

#import "DoubleSeries.h"

@implementation DoubleSeries {
    SCIArrayController *xArray;
    SCIArrayController *yArray;
}

- (SCIGenericType)xValues {
    SCIGenericType arrayPointer;
    arrayPointer.voidPtr = xArray.data;
    arrayPointer.type = SCIDataType_DoublePtr;
    
    return arrayPointer;
}

- (SCIGenericType)yValues {
    SCIGenericType arrayPointer;
    arrayPointer.voidPtr = yArray.data;
    arrayPointer.type = SCIDataType_DoublePtr;
    
    return arrayPointer;
}

- (int)size {
    return [xArray count];
}

- (instancetype)init
{
    self = [super init];
    if (self) {
        xArray = [[SCIArrayController alloc] initWithType:SCIDataType_Double];
        yArray = [[SCIArrayController alloc] initWithType:SCIDataType_Double];
    }
    return self;
}

- (instancetype)initWithCapacity: (int) capacity {
    self = [super init];
    if (self) {
        xArray = [[SCIArrayController alloc] initWithType:SCIDataType_Double Size:capacity];
        yArray = [[SCIArrayController alloc] initWithType:SCIDataType_Double Size:capacity];
    }
    return self;
}

- (void)addX: (double)x Y:(double)y {
    [xArray append:SCIGeneric(x)];
    [yArray append:SCIGeneric(y)];
}

@end

File: v2.x/Examples/SciChartSwiftDemo/SciChartSwiftDemo/ImportantFiles/DoubleSeries.swift
//
//  DoubleSeries.swift
//  SciChartSwiftDemo
//
//  Created by Admin on 19/04/2017.
//  Copyright © 2017 SciChart Ltd. All rights reserved.
//

import UIKit
import SciChart

class DoubleSeries: NSObject {
    
    private var xArray: SCIArrayController
    private var yArray: SCIArrayController
    
    var xValues: SCIGenericType {
        var arrayPointer = SCIGeneric(xArray.data())
        arrayPointer.type = .doublePtr        
        return arrayPointer
    }
    
    var yValues: SCIGenericType {
        var arrayPointer = SCIGeneric(yArray.data())
        arrayPointer.type = .doublePtr
        return arrayPointer
    }
    
    var size: Int32 {
        return xArray.count()
    }
    
    override init() {
        xArray = SCIArrayController(type: .double)
        yArray = SCIArrayController(type: .double)
        super.init()
    }

    init(capacity: Int32) {
        xArray = SCIArrayController(type: .double, size: capacity)
        yArray = SCIArrayController(type: .double, size: capacity)
        super.init()
    }
    
    func addX(_ x: Double, y: Double) {
        xArray.append(SCIGeneric(x))
        yArray.append(SCIGeneric(y))
    }
}
